### PR TITLE
Filter by parameters

### PR DESCRIPTION
**Pull request by ShacharOr100**

### CODE REVIEW

File: src/controllers/ArticleController.ts [hunk 1]
```diff
@@ -29,7 +31,51 @@ export class ArticleController extends AbstractController {
                 return res.status(200).json(answer)
             }
         });
+    }
+
+    public filterArticleByTitle = async(req:any, res:any) => {
+        let partialTitleName = req.params["_partialTitleName"]
+        if (!partialTitleName) {
```
Comment (guyeise5) [inline src/controllers/ArticleController.ts]: Unnecessary if statement

File: src/controllers/ArticleController.ts [hunk 2]
```diff
@@ -29,7 +31,51 @@ export class ArticleController extends AbstractController {
...
+            res.status(404).json("Could not find article with given partial title")
+        } else {
+            let partialTitleMatchRegex = ('^' + partialTitleName + '*').toString()
```
Comment (guyeise5) [inline src/controllers/ArticleController.ts]: replace '^' with '*'

Comment (ShacharOr100) [reply src/controllers/ArticleController.ts]: It's a regex, it's not sql LIKE.

File: src/controllers/ArticleController.ts (hunk 1 above)
Comment (ShacharOr100) [reply src/controllers/ArticleController.ts]: you are right, if the title parameter were not in route that function would not be invoked.

**Pull request closed without merging**